authors/apps/comments/views.py

```diff
@@ -0,0 +1,150 @@
+from ..profiles.models import Profile
+from .models import Comment, Reply
+from ..articles.models import Article
+from rest_framework.response import Response
+
+
+class CommentsAPIView(generics.GenericAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentRenderer,)
+
+    def get(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        comments = Comment.objects.filter(article=article)
+        serialize_data = self.serializer_class(comments, many=True)
+        return Response({"comments": serialize_data.data}, status=status.HTTP_200_OK)
+
+    def post(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        data = request.data
+        author = Profile.objects.get(user=request.user)
+        serializer = self.serializer_class(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(
+            author=author,
+            article=article
+        )
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CommentDetailsAPIView(generics.GenericAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentRenderer,)
+
+    def required_objects(self, request, pk):
+        self.comment = get_object_or_404(Comment, pk=pk)
+        self.data = request.data
+
+    def author_requirements(self, request):
+        self.requester = Profile.objects.get(user=request.user)
+        self.valid_author = self.comment.author == self.requester
+
+    def get(self, request, pk):
+        """gets one comment and its details"""
+        comment = get_object_or_404(Comment, pk=pk)
+        serialize_data = self.serializer_class(comment)
+        return Response(serialize_data.data, status=status.HTTP_200_OK)
+
+    def patch(self, request, pk):
+        self.required_objects(request, pk)
+        self.author_requirements(request)
+
+        if not self.valid_author:
+            return Response({"response": "you can only update comments you made"},
+                            status=status.HTTP_403_FORBIDDEN)
+
+        serializer = self.serializer_class(
+            self.comment, self.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def delete(self, request, pk):
+        self.required_objects(request, pk)
+        self.author_requirements(request)
+
+        if not self.valid_author:
+            return Response({"response": "you can only delete comments you made"},
+                            status=status.HTTP_403_FORBIDDEN)
+        self.comment.delete()
+        return Response({
+            "message": "Successfully deleted comment"},
+            status=status.HTTP_200_OK)
+
+
+class ReplyAPIView(generics.GenericAPIView):
```

Comment:
@sebbss, be specific with the `generic view you trying to use` For example for `post` and `get` , you can use ListCreateAPIView.
More details on https://www.django-rest-framework.org/api-guide/generic-views